### ISSUE TITLE
batch_norm const running_mean/_var modified in-place

### ISSUE THREAD

**Issue by cpuhrsch** [pytorch/pytorch#46808]
## 🐛 Bug

batch_norm modifies [running_mean](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Normalization.cpp#L260) and [running_var](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Normalization.cpp#L264) in-place and yet [the schema](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L743) indicates the arguments are const (they are marked as "Tensor?") and so does [the function signature](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Normalization.cpp#L493). 

## To Reproduce

This currently causes no bugs within PyTorch, because all code [uses the underlying pointers](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Normalization.cpp#L230). However, when overwriting this function for a different dispatch key (as maybe for nestedtensor) in-place modification of running_mean/_var is prohibited.

## Expected behavior

running_mean/_var should not be const.

## Environment

N/A

## Additional context

This can be currently worked around via unsafe Tensor operations that ignore const.

cc @ezyang @gchanan @zou3519 @bdhirsh @heitorschueroff @albanD @mruberry

**Comment by SsnL** [comment]
see also https://github.com/pytorch/pytorch/issues/13402

**Comment by mruberry** [comment]
Linking other proposed batchnorm changes: https://github.com/pytorch/pytorch/issues/39674.

**Comment by ezyang** [comment]
> However, when overwriting this function for a different dispatch key (as maybe for nestedtensor) in-place modification of running_mean/_var is prohibited.

This is not actually true; you can still use inplace ops on a `const Tensor&` reference.

**Comment by albanD** [comment]
This is a known issue.
Closing in favor of the original issue that has a lot of discussion already: https://github.com/pytorch/pytorch/issues/13402

**Comment by cpuhrsch** [comment]
> > However, when overwriting this function for a different dispatch key (as maybe for nestedtensor) in-place modification of running_mean/_var is prohibited.
> 
> This is not actually true; you can still use inplace ops on a `const Tensor&` reference.

Sure, but should you?

**Comment by ezyang** [comment]
@cpuhrsch yes it is by design. See https://github.com/zdevito/ATen/issues/27
